### PR TITLE
Check all parent classes in extendsAbstract(), not only one direct parent

### PR DESCRIPTION
**Pull request by alfredbez**
## PR Description

(From #320)
When checking if an annotation should be added or not, there is a check for the direct parent class. This does not take all parent classes into consideration. We regularly extend some Plugin from the `Spryker` namespace in our `Pyz` namespace and therefore do not extend the `AbstractPlugin`, but the `FooBarPlugin` class in the `Spryker` namespace, that extends the `AbstractPlugin` itself. I extended the check from the direct parent of a class, to check against all parent classes.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php [hunk 1]
```diff
@@ -33,9 +34,49 @@ protected function extendsAbstract(File $phpCsFile, int $stackPointer, string $a
...
+        }
+
         return false;
     }
 
+    /**
+     * @param \PHP_CodeSniffer\Files\File $phpCsFile
+     * @param int $stackPointer
+     *
+     * @return array
```
Comment (dereuromark) [inline Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php:48]: These are `array<string>` right?

Comment (alfredbez) [reply Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php]: Yes, I wasn't sure, if I can use that already in this repo, I've updated it now.

File: Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php [hunk 2]
```diff
@@ -33,9 +34,49 @@ protected function extendsAbstract(File $phpCsFile, int $stackPointer, string $a
             return true;
         }
 
+        if (in_array($abstractName, $this->getParentClassesFor($phpCsFile, $stackPointer))) {
```
Comment (dereuromark) [inline Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php:37]:
```suggestion
        if (in_array($abstractName, $this->getParentClassesFor($phpCsFile, $stackPointer), true)) {
```

**Comment by dereuromark** [comment]
Seems like PHPStan has one issue left
```
  Line   Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php  
 ------ ---------------------------------------------------------------------- 
  50     PHPDoc tag @param references unknown parameter: $phpCsFile            
  66     Parameter #1 $argument of class ReflectionClass constructor expects   
         class-string<T of object>|T of object, string given.                  
 ```

**Pull request merged**